tests/unit-tests/lib/test_str_util.cpp

```diff
@@ -39,58 +39,317 @@ namespace test_str_util {
 
     // Tests that Foo does Xyz.
 
-    TEST_F(test_str_util, path_to_filename) {
-        std::string fname = "";
-        ASSERT_EQ(path_to_filename("/home/blah", fname), 0);
-        ASSERT_EQ(fname, "blah");
-        ASSERT_EQ(path_to_filename("hellokeith", fname), 0);
-        ASSERT_EQ(fname, "hellokeith");
-        fname = "";
-        ASSERT_EQ(path_to_filename("/home/blah/", fname), -2);
-        ASSERT_EQ(fname, "");
-        ASSERT_EQ(path_to_filename("", fname), -1);
-        ASSERT_EQ(fname, "");
+    TEST_F(test_str_util, ndays_to_string) {
+        char buf[128];
+        char *nilbuf {0};
+        EXPECT_TRUE(ndays_to_string(-1.0, 0, buf));
+        EXPECT_FALSE(ndays_to_string(0.0, -1, buf));
+        EXPECT_STREQ(buf, "");
```

Comment:
Can you add these?
`EXPECT_TRUE(ndays_to_string(5, -1, buf));`
`EXPECT_STREQ(buf, "5 sec ");`

Reply:
I adjusted this so the tests don't fail. It's `5 days ` and the function returns `false` on success.